tests/unit/checkmate/checker/pipeline/core_test.py

```diff
@@ -0,0 +1,57 @@
+import os
+from unittest.mock import create_autospec, sentinel
+
+import pytest
+
+from checkmate.checker.pipeline.core import Pipeline, Stage
+from checkmate.exceptions import StageException
+
+
+class TestStage:
+    def test_call_must_be_overriden(self):
+        with pytest.raises(NotImplementedError):
+            Stage()(working_dir="example")
+
+    def test_temp_file(self, tmpdir):
+        temp_file = Stage.temp_file(tmpdir, suffix=".suffix")
+
+        assert temp_file.name.startswith(str(tmpdir))
+        assert temp_file.name.endswith(".suffix")
+        temp_file.write(b"content")  # Writable!
+
+        # Persists on close
+        temp_file.close()
+        assert os.path.exists(temp_file.name)
+
+        # Has the content
+        with open(temp_file.name) as handle:
+            content = handle.read()
+
+        assert content == "content"
```

Comment:
If there's a shorter way of saying: "It gives you an open writable file that doesn't delete itself magically", I'd like to know